code/go/0chain.net/smartcontract/zcnsc/authorizer.go

```diff
@@ -336,6 +337,28 @@ func (zcn *ZCNSmartContract) UpdateAuthorizerConfig(
 		return "", common.NewError(code, err.Error())
 	}
 
+	sp, err := zcn.getStakePool(in.ID, ctx)
+	if err != nil {
+		if err == util.ErrValueNotPresent {
+			msg := fmt.Sprintf("update not allowed due to missing stakepool details for (authorizerID: %v), err: %v", authorizer.ID, err)
+			err = common.NewError(code, msg)
+			Logger.Error("updating settings", zap.Error(err))
+			return "", err
+		}
+		msg := fmt.Sprintf("unexpected error for (authorizerID: %v), err: %v", authorizer.ID, err)
+		err = common.NewError(code, msg)
+		Logger.Error("updating settings", zap.Error(err))
+		return "", err
+	}
+
+	// Do not allow update authorizer in absence of delegate wallet
+	if sp.Settings.DelegateWallet == "" {
```

Comment:
We better check if ``sp.Settings.DelegateWallet == t.ClientID`` so only the manager of provider could update it